webapp/handlers.py

```diff
@@ -0,0 +1,94 @@
+import socket
+from urllib.parse import unquote, urlparse, urlunparse
+
+import flask
+import webapp.template_utils as template_utils
+
+
+def set_handlers(app):
+    @app.context_processor
+    def utility_processor():
+        """
+        This defines the set of properties and functions that will be added
+        to the default context for processing templates. All these items
+        can be used in all templates
+        """
+
+        return {
+            # Functions
+            "format_date": template_utils.format_date,
+            "get_json_feed": template_utils.get_json_feed_content,
+            "get_year": template_utils.get_year,
```

Comment:
How about:

``` python
"current_year": datetime.now().year
```

?

Reply:
Could do. I prefer putting it into `template_utils` though. Its a bit more passing through, but makes it more clearly organized.

Reply:
you're masking a simple `datetime.now()` behind a function call. You're also calling a function when a static function would do perfectly fine.

I'm not super opinionated about where you put it, but I do think it's both inelegant and inefficient to call a function for this one very short line of code.

Reply:
Thats true, my thought here is more on maintainability. Since all template utilities have their logic in `template_utils.py` it seems like the place to look for them. Which is the first place I would look. 
Not finding a function here would lead me to more searching.
Maybe mapping them in `handlers.py` is a pitfall? 

The performance overhead are 2 explicit function calls, which in turn could call further things in the `datetime` module. 
If the runtime gets warm this will probably be runtime optimized as well? 

The only downside of having a static value seems to be an outdated year if deployment for a long running app happens. This could lead to a bug on year changes.

There might be something I am missing here though.

Reply:
I'm not worries about performance, I'm also worried about maintainability. The trouble is that you would look in `template_utils` because you wrote the application and know the structure. If I didn't write the application, I instead would start from the template itself, and have to jump one extra step. But even in your structure you could at least have done `"current_year": template_utils.get_year()` to simplify what's passed to the template.

Anyway, maybe you could cover both our concerns by doing:

``` python3
    app.context_processor(template_utils.utility_processor)
```

However, I'm not super precious. If it would delay things too much for you to change this, you can merge without it. I'll happily add a follow-up PR.